### DISCUSSION TITLE
File upload issue

### DISCUSSION THREAD

**Discussion by rabol** [spatie/laravel-livewire-wizard#82] [Q&A]
Hi

I have a wizard where user kan upload a file.

After the use goes to the next page, if he then press the navigation link to go to the previous page, then the validation of the file type fails, no matter what file is selected.

same issues if I have a 'Back' button and call $this->previousStep();

so somehow the file validation is screwed up as soon as you go to next page

I have also noticed that the mount() is called on 'previous', so maybe the state is corrupt?

Any hint on how to fix this ?

**Comment by mawaterq** [comment]
@rabol Did you find a way for this? I am facing the same issue. When I go to next step as the file belongs to a user (user must be created first), I need to do everything is the final step. The file value is gone in the last step but I can get it in the step that contains the file upload fields.

**Comment by rabol** [comment]
@mawaterq In my case I saved the information about the uploaded file(s) in an array of the state and then processed them in the summary step.

then in the next / prev situation I 'displayed' the file information and if the user then do a new upload I delete the old

**Reply by mawaterq** [reply]
Thanks for your reply! Will do that as it make more sense in my case as well.
